scipy/stats/morestats.py

```diff
@@ -1160,7 +1208,13 @@ def _all(x, brack):
         raise ValueError("Method %s not recognized." % method)
 
     optimfunc = methods[method]
-    return optimfunc(x, brack)
+    try:
+        return optimfunc(x)
+    except AttributeError as e:
+        if "has no attribute 'x'" in str(e):
+            raise ValueError("`optimizer` must return an `OptimizeResult` object")
+        else:
+            raise
```

Comment:
It may just be a style preference, but branching based on the error message string makes me uncomfortable. I think I'll change the way an exception gets raised to avoid this.